Kfold cross validation using cross_val_score to determine the best model for iris dataset.

In [7]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
import numpy as np

In [8]:
iris = load_iris()

LOGISTIC REGRESSION

In [9]:
l_scores = cross_val_score(LogisticRegression(), iris.data, iris.target)
l_scores

/usr/local/lib/python3.11/dist-packages/sklearn/linear_model/_logistic.py:465: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [10]:
np.average(l_scores)

np.float64(0.9733333333333334)

DECISION TREE

In [11]:
d_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
d_scores

array([0.96666667, 0.96666667, 0.9       , 1.        , 1.        ])

In [12]:
np.average(d_scores)

np.float64(0.9666666666666668)

SUPPORT VECTOR MACHINE (SVM)

In [13]:
s_scores = cross_val_score(SVC(), iris.data, iris.target)
s_scores

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

In [14]:
np.average(s_scores)

np.float64(0.9666666666666666)

RANDOM FOREST

In [15]:
r_scores = cross_val_score(RandomForestClassifier(n_estimators=40), iris.data, iris.target)
r_scores

array([0.96666667, 0.96666667, 0.9       , 0.93333333, 1.        ])

In [16]:
np.average(r_scores)

np.float64(0.9533333333333334)

# **Best score so far is from Logistic Regression: 0.9733333333333334**